infra/src/main/kotlin/fi/fta/geoviite/infra/switchLibrary/SwitchStructure.kt

```diff
@@ -266,6 +274,24 @@ data class SwitchStructure(
     fun getJointLocation(jointNumber: JointNumber): Point {
         return getJoint(jointNumber).location
     }
+
+    fun stripUniqueIdentifiers(): SwitchStructure {
+        var i = 0;
+        return copy(
+            id = IntId(i++),
+            alignments = alignments.map { a ->
+                a.copy(
+                    id = IntId(i++),
+                    elements = a.elements.map { e ->
+                        when (e) {
+                            is SwitchElementLine -> e.copy(id= IntId(i++))
+                            is SwitchElementCurve -> e.copy(id= IntId(i++))
+                        }
+                    }
+                )
+            }
+        )
+    }
```

Comment:
Tässä on vissiin tarkoituksena että korvataan kaikki vaihderakenteen ja sen sisätyyppien id-kentät jollain sillä hetkellä generoiduilla? Sen sijasta että käyttäisi juoksevaa numerointia, niin voisikohan ne vain korvata nollilla? 🤔 Tällöin ei tarvisi kuljettaa tuota `i`:tä omassa muuttujassaan ja päivitellä sitä

Reply:
:thinking: Toimisi varmaan yhtä hyvin, koska tätähän käytetään vain vertailuun. Sinällään tuon vertailufunktion (nyk `areSame`) voisi siirtää tänne nimelle `isSame(other: SwitchStructure): Boolean`, jolloin tämän koko `stripIdentifiers` voi muuttaa privateksi. Korostaa sitä että lopputuloksena tulee puppu-id:t sisältävä olio ja privaten ansiosta sitä ei pääse vahingossakaan käyttämään missään muualla kun vertailussa itsessään.

Reply:
Juoksevia IDeitä täytyy käyttää, koska mallissa on tarkastus elementtien yksilöllisille IDeille. Tuo `isSame` on joo hyvä ajatus.